### PR TITLE
Fix J9VMInternals.getStackTrace() to return null for a null throwable

### LINKED ISSUE #15937: Loom: failure in GetStackTraceALot
see https://github.com/eclipse-openj9/openj9/issues/15152#issuecomment-1251743982

### PR DESCRIPTION
**Pull request by thallium**
A null throwable will be passed when a virtual thread is during
transition, in this case we simply return null and the caller
VirtualThread.asyncGetStackTrace() will treat a null return as retry and
other callers are not affected.

Also minor fix for coding style

Fixes #15937
FYI @fengxue-IS 

Signed-off-by: Gengchen Tuo <gengchen.tuo@ibm.com>

### CODE REVIEW

File: runtime/jcl/common/java_lang_J9VMInternals.cpp [hunk 1]
```diff
@@ -32,14 +32,17 @@
 extern "C" {
 
 jobject JNICALL
-Java_java_lang_J9VMInternals_getStackTrace(JNIEnv * env, jclass recv, jobject throwable, jboolean pruneConstructors)
+Java_java_lang_J9VMInternals_getStackTrace(JNIEnv *env, jclass recv, jobject throwable, jboolean pruneConstructors)
 {
-	J9VMThread *currentThread = (J9VMThread*)env;
+	if (NULL == throwable) {
+		return NULL;
+	}
```
Comment (fengxue-IS) [inline runtime/jcl/common/java_lang_J9VMInternals.cpp:39]: Please put this under Java 19 ifdef, for pre-Java 19, it is not possible for a null throwable to be passed into this method.

**Review by fengxue-IS — CHANGES_REQUESTED** [review]

File: runtime/jcl/common/java_lang_J9VMInternals.cpp (hunk 1 above)
Comment (thallium) [reply runtime/jcl/common/java_lang_J9VMInternals.cpp:40]: done

**Review by fengxue-IS — APPROVED** [review]
LGTM
@babsingh please take a look.

**Comment by babsingh** [comment]
More details are required:
- Why null throwable is being passed to J9VMInternals.getStackTrace?
- Will the new behaviour negatively impact other callers of J9VMInternals.getStackTrace?
- Is the new behaviour comparable to RI?
- Which tests are needed to fully verify this PR?

**Comment by fengxue-IS** [comment]
> * Why null throwable is being passed to J9VMInternals.getStackTrace?

https://github.com/eclipse-openj9/openj9/pull/15846 patched `Java_java_lang_Thread_getStackTraceImpl` to support virtual thread, an extra check was added for virtual thread where if the thread is during transition (mount/unmount), then it will return null directly instead of continuing. The caller [VirtualThread.asyncGetStackTrace()](https://github.com/ibmruntimes/openj9-openjdk-jdk19/blob/openj9/src/java.base/share/classes/java/lang/VirtualThread.java#L854) will treat a null return as retry.
> * Will the new behaviour negatively impact other callers of J9VMInternals.getStackTrace?

No, since all other callers of J9VMInternals.getStackTrace should be suppling a valid throwable object same as before.
> * Is the new behaviour comparable to RI?

Based on code inspection, Yes.
> * Which tests are needed to fully verify this PR?

This is a timing based issue and the only test currently in place that could verify this is JTReg: `java/lang/Thread/virtual/stress/GetStackTraceALot.java` (The test which this PR targets to fix)

**Comment by babsingh** [comment]
@thallium The above details should be documented in the commit message, and a comment should be added in the code highlighting it as a virtual thread specific scenario.

**Comment by thallium** [comment]
> @thallium The above details should be documented in the commit message, and a comment should be added in the code highlighting it as a virtual thread specific scenario.

Done

File: runtime/jcl/common/java_lang_J9VMInternals.cpp [hunk 2]
```diff
@@ -32,14 +32,21 @@
 extern "C" {
 
 jobject JNICALL
-Java_java_lang_J9VMInternals_getStackTrace(JNIEnv * env, jclass recv, jobject throwable, jboolean pruneConstructors)
+Java_java_lang_J9VMInternals_getStackTrace(JNIEnv *env, jclass recv, jobject throwable, jboolean pruneConstructors)
 {
-	J9VMThread *currentThread = (J9VMThread*)env;
+#if JAVA_SPEC_VERSION >= 19
+	/* throwable can only be null for a virtual thread that is during transition */
```
Comment (babsingh) [inline runtime/jcl/common/java_lang_J9VMInternals.cpp:38]:
```suggestion
	/* throwable can only be null for a virtual thread during mount and unmount transitions. */
```

**Review by babsingh — CHANGES_REQUESTED** [review]
Also, the PR description should reflect the latest commit message.

File: runtime/jcl/common/java_lang_J9VMInternals.cpp (hunk 2 above)
Comment (babsingh) [reply runtime/jcl/common/java_lang_J9VMInternals.cpp]: `that is` was removed in the suggestion.

Comment (babsingh) [reply runtime/jcl/common/java_lang_J9VMInternals.cpp]: still missing the punctuation `.` (full-stop)

**Review by babsingh — COMMENTED** [review]
Approved. LGTM.

**Pull request merged**